core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -15,104 +15,137 @@
  * This is a fluent logger helper class that wraps a plug-able {@link Logger}.
  *
  * <p>This logger logs format-able messages that use {@code {}} as the placeholder. When a throwable is the last
- * argument of the format varargs and the logger is enabled for {@link ServiceLogger#asVerbose() verbose} logging the
+ * argument of the format varargs and the logger is enabled for {@link ClientLogger#asVerbose() verbose} logging the
  * stack trace for the throwable will be included in the log message.</p>
  *
  * <p>A minimum logging level threshold is determined by the {@link BaseConfigurations#AZURE_LOG_LEVEL AZURE_LOG_LEVEL}
  * environment configuration, by default logging is disabled. The default logging level for messages is
- * {@link ServiceLogger#asInfo() info}.</p>
+ * {@link ClientLogger#asInfo() info}.</p>
  *
  * <p><strong>Log level hierarchy</strong></p>
  * <ol>
- *     <li>{@link ServiceLogger#asError() Error}</li>
- *     <li>{@link ServiceLogger#asWarning() Warning}</li>
- *     <li>{@link ServiceLogger#asInfo() Info}</li>
- *     <li>{@link ServiceLogger#asVerbose() Verbose}</li>
+ * <li>{@link ClientLogger#asError() Error}</li>
+ * <li>{@link ClientLogger#asWarning() Warning}</li>
+ * <li>{@link ClientLogger#asInfo() Info}</li>
+ * <li>{@link ClientLogger#asVerbose() Verbose}</li>
  * </ol>
  *
  * @see Configuration
  */
-public class ServiceLogger implements ServiceLoggerAPI {
-    private static final NoopServiceLogger NOOP_LOGGER = new NoopServiceLogger();
+public class ClientLogger {
+    private static final NoopClientLogger NOOP_LOGGER = new NoopClientLogger();
 
     private final Logger logger;
 
+    /**
+     * Indicate that log level is at trace level.
+     */
+    public static final int TRACE_LEVEL = 0;
```

Comment:
The logging levels should be private to the logger, users don't need to know about them as they are meant for internal tracking reasons.